src/index.ts

```diff
@@ -187,6 +188,10 @@ await server.register(proxy, {
     .map((module) => module.trim()),
 });
 
-server.listen({ port: 3000 }).then((res) => {
+const port = Number.isNaN(Number(process.env.PORT))
+  ? 3000
+  : Number(process.env.PORT);
```

Comment:
Will TS be happy if you do this?
```suggestion
const port = Number(process.env.PORT) || 3000;
```

Reply:
Yep, TS is happy with that!